presto-hive/src/main/java/io/prestosql/plugin/hive/HiveConfig.java

```diff
@@ -165,6 +165,8 @@
     private long fileStatusCacheMaxSize = 1000 * 1000;
     private List<String> fileStatusCacheTables = ImmutableList.of();
 
+    private Duration hiveTxnHeartBeatInterval = new Duration(5, TimeUnit.SECONDS);
```

Comment:
Rather than having users to configure this, can we ask transaction manager how often it requires the heartbeat?
We still may have the config as an override.

Reply:
Also

- don't abbreviate
- we use "heartbeat" as a single word

```suggestion
    private Duration hiveTransactionHeartbeatInterval = new Duration(5, TimeUnit.SECONDS);
```

Reply:
Hive has a config of defining the period for which if no heartbeat is received, the transaction is aborted. I can derive heartbeat interval from that, Hive is also doing that: https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/lockmgr/DbTxnManager.java#L927

Reply:
@stagraqubole i didn't mean to read Hive config although it's a reasonable fallback if `hive.config.resources` is set. I don't want to _force_ users to use `hive.config.resources` though.

What I meant, however, is that we could perhaps _ask_ HMS what's it timeout setting.
Why would user need to configure timeouts on every host when configuring HMS could be enough?
(From cursory reading of hive code, i didn't find how to do this, but i didn't explore all the options.)

Reply:
Found a thrift call "get_config_value". I can get the value of "hive.txn.timeout" from the Metastore server over thrift and derive heartbeat interval from it when not defined by user.